packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1610,9 +1611,12 @@ export class Bucket extends BucketBase {
     }
     Bucket.validateBucketName(bucketName);
 
-    const newUrlFormat = attrs.bucketWebsiteNewUrlFormat === undefined
-      ? false
-      : attrs.bucketWebsiteNewUrlFormat;
+    const oldUrlFormatRegions = new Set(regionsBefore(RULE_S3_WEBSITE_REGIONAL_SUBDOMAIN));
+
+    const newUrlFormat = !oldUrlFormatRegions.has(region) ||
+      (attrs.bucketWebsiteNewUrlFormat === undefined
+        ? false
+        : attrs.bucketWebsiteNewUrlFormat);
```

Comment:
The other thing here is that I believe this would change the behavior of an explicitly set `bucketWebsiteNewUrlFormat`? I think the best way would be to honor any explicitly set value and only change the default behavior to auto detect.

With this new feature, I'm also tempted to deprecate `bucketWebsiteNewUrlFormat`. With the `region` and explicit `bucketWebsiteUrl` attributes, there's plenty of ways to influence the behavior. What do you think?

Reply:
Yes I agree, I was wondering why someone should set `bucketWebsiteNewUrlFormat` explicitly to true. I guess because one is aware of the url format differences between some regions.

I think, mark `bucketWebsiteNewUrlFormat` as deprecated, and only change the default behaviour with this feature as you suggested is fine 👌

Reply:
I changed the logic, so that the old behaviour is applied if `bucketWebsiteNewUrlFormat` is explicitly specified. I also made `bucketWebsiteNewUrlFormat` deprecated with a short description of the new behaviour and that `bucketWebsiteNewUrlFormat` still affects the url format if specified though. I am not sure if we need also updated doc strings for `bucketWebsiteUrl` and `region`.